File: dotnet/test/HelloWorldTests.cs
namespace Hello.World;

public class HelloWorldTests
{
    [Fact]
    public void PrintGreetingWithoutNameTest()
    {
        string[] args = new string[] { };
        string expectedOutput = "Hello World from Dotnet !";
        Assert.Equal(expectedOutput, HelloWorld.PrintGreeting(args));
    }

    [Fact]
    public void PrintGreetingWithNameTest()
    {
        string[] args = new string[] { "Alter", "Sabeh" };
        string expectedOutput = "Hi Alter Sabeh, greetings from Dotnet !";
        Assert.Equal(expectedOutput, HelloWorld.PrintGreeting(args));
    }

    [Fact]
    public void PrintDateTest()
    {
        DateTime now = DateTime.Now;
        string formattedDate = now.ToString("MMMM d, yyyy HH:mm");
        string expectedOutput = $"Today is {formattedDate}";
        Assert.Equal(expectedOutput, HelloWorld.PrintDate());
    }
}

File: swiftpm/Tests/HelloWorldTests/HelloWorldTests.swift
import XCTest

@testable import HelloWorld

final class HelloWorldTests: XCTestCase {
  func testPrintGreetingWithoutName() {
    let args: [String] = []
    let expectedOutput = "Hello World from SwiftPM !"
    let actualOutput = printGreeting(args)
    XCTAssertEqual(actualOutput, expectedOutput)
  }

  func testPrintGreetingWithName() {
    let args = ["", "Alter", "Sabeh"]
    let expectedOutput = "Hi Alter Sabeh, greetings from SwiftPM !"
    let actualOutput = printGreeting(args)
    XCTAssertEqual(actualOutput, expectedOutput)
  }

  func testPrintDate() {
    let now = Date()
    let formattedDate = DateFormatter.localizedString(
      from: now, dateStyle: .long, timeStyle: .short
    )
    let expectedOutput = "Today is \(formattedDate)"
    let actualOutput = printDate()
    XCTAssertEqual(actualOutput, expectedOutput)
  }
}
